crates/hotshot/src/lib.rs

```diff
@@ -641,14 +651,24 @@ impl<TYPES: NodeType> HotShotInitializer<TYPES> {
         Ok(Self {
```

Comment:
I can't comment on the line but while we're here, can we make `from_genesis` ^ take `instance_state` by value to avoid unnecessary clone and be consistent with `from_reload`?